### ISSUE TITLE
Simplify the date field (in Symphony 3)?

### ISSUE THREAD

**Issue by michael-e** [symphonycms/symphony-2#2492]
In #2491, there was a long discussion about certain data being "dates" or not. I myself came to this conclusion:

> Thinking about it a bit more, even the possibility to not display the time in the current Date Field is terribly dangerous. When inputting a date without time, I wouldn't expect this to get localized. But exactly this will happen.

> So I tend to say: In Symphony 3, remove the ability to not display the time. It is misleading. If you only need year/month/date (or only a year), you should not use a date field, because all you have is numbers or strings (unlocalized).

**Comment by jensscherbl** [comment]
I'd just have separate `Date` and `Datetime` fields, with `Datetime` being stored in UTC internally and an option (checked by default) to localize or not for output and comparison. Makes sense?

**Comment by michael-e** [comment]
And the Date field? Shouldn't it get the option as well? Or should it never be loccalized?

**Comment by jensscherbl** [comment]
> And the Date field? Shouldn't it get the option as well? Or should it never be loccalized?

Yeah, just edited my comment.

**Comment by michael-e** [comment]
Ah, that makes sense.

**Comment by nitriques** [comment]
> I'd just have separate Date and Datetime fields.

I think this is not a good idea. timestamps always contains seconds and it's only a mater of formatting. (input formatting and output formatting)

**Comment by jensscherbl** [comment]
> I think this is not a good idea. timestamps always contains seconds and it's only a mater of formatting. 

Not sure what timestamps have to do with this. Can you explain?

**Comment by nitriques** [comment]
All dates are timestamps. Even if is a *date only* field, the timestamp will contains the hours, minutes and seconds. So 2015-08-28 (no time) is actually 2015-08-28 00:00:00.000

**Comment by jensscherbl** [comment]
Ah yes. That's exactly what we wanted to solve with a separate `Date` and `Datetime` fields. A `Date` field would actually be just a date, not a timestamp.

**Comment by nitriques** [comment]
How would you store it ? It needs to be stored as a timestamp (MySQL datetime is a timestamp). Saving it as a string is not a good idea IMHO.

**Comment by jensscherbl** [comment]
> How would you store it?

As a [`DATE`][1]?

> It needs to be stored as a timestamp

Why?

> MySQL datetime is a timestamp

Are you sure? Maybe for MySQL internally, but according to the [documentation][1], these are separate types with different behaviour.

[1]: https://dev.mysql.com/doc/refman/5.5/en/datetime.html

**Comment by nitriques** [comment]
> Maybe for MySQL internally,

Yes that's what I meant. DATE is a truncated timestamp (for saving storage space). But since hard drives are so cheap, I do not see the point of using the DATE type.

But if you guys need it, yes we could create separate field.

**Comment by michael-e** [comment]
No I don't need it. Just thought that it might make things a bit more clear. But maybe it won't. :-(

**Comment by nitriques** [comment]
> Just thought that it might make things a bit more clear.

It may, but for me, it's not maintainable

1. Lots of code duplication (unless both fields are provided by the same extension...?)
2. I cannot change the field easily to accept time in the future.

I really see the validator more useful. And as you said, it can me backward compatible.

We would only need to do something like Nils and I did https://github.com/hananils/datetime/blob/master/fields/field.datetime.php#L574-L582

**Comment by michael-e** [comment]
Sounds reasonable to me.

**Comment by jensscherbl** [comment]
> But since hard drives are so cheap, I do not see the point of using the DATE type.

The point is not storing a time when you don't want to store a time.

> I cannot change the field easily to accept time in the future.

Same for other fields. You can also not easily change a number field to accept text in the future.
